rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -243,7 +255,7 @@ bool TypeSupport::serializeROSmessage(eprosima::fastcdr::Cdr &ser, const rosidl_
                     break;
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING:
                     {
-                        std::string &str = *(std::string*)field;
+                        std::string && str = StringHelper<MembersType>::convert_to_std_string(field);
```

Comment:
This was a leftover from a bad rebase I think. Anyway, I've updated this to use the universal reference and `auto`, and `StringHelper::convert_to_std_string` to return a reference. The idea is that `StringHelper` does not create an extra copy of `std::string` for C++ typesupport, and by using `auto &&` it can be bound as an lvalue as well.